### PR TITLE
Best chain improvements

### PR DESCRIPTION
**Pull request by Isaac-DeFrain**
These are commits independent of and cherry-picked from the `snapshot` branch.

- adds a default `--num 10` for the `best-chain` command
- improves the `best-chain` calculation by fetching blocks from the db
- uses `Display` over `Debug` for displaying `IndexerState` and `Branch`
- updates README with correct CLI flags

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,26 +37,24 @@ Build (install [nix](#about-the-development-environment) first)
 
 ```sh
 nix develop
-nix build '.?submodules=1'
+nix build
```
Comment (jenr24G) [inline README.md:40]: did you fix this so `'.?submodules=1'` is no longer necessary?

File: src/server/mod.rs [hunk 2]
```diff
@@ -309,12 +309,14 @@ async fn handle_conn(
             let data_buffer = buffers.next().unwrap();
             let num = String::from_utf8(data_buffer[..data_buffer.len() - 1].to_vec())?
                 .parse::<usize>()?;
-            let best_chain: Vec<PrecomputedBlock> = best_chain[..best_chain.len() - 1]
-                .iter()
-                .take(num)
-                .cloned()
-                .map(|state_hash| db.get_block(&state_hash).unwrap().unwrap())
-                .collect();
+            let mut parent_hash = best_tip.parent_hash;
```
Comment (jenr24G) [inline src/server/mod.rs:312]: nice! less data going in to `handle_conn`

File: src/state/branch.rs [hunk 3]
```diff
@@ -85,7 +74,60 @@ impl Branch {
         self.branches.height() == 0
     }
 
-    /// Returns the new node's id in the branch
+    /// Returns the node id of the best tip
+    pub fn best_tip_id(&self) -> NodeId {
```
Comment (jenr24G) [inline src/state/branch.rs:78]: did this already exist on the branch class?

File: src/state/mod.rs [hunk 4]
```diff
@@ -431,6 +431,7 @@ impl IndexerState {
         // now add the successive non-canoical blocks
```
Comment (jenr24G) [inline src/state/mod.rs:431]: typo!

File: src/state/mod.rs [hunk 5]
```diff
@@ -976,16 +977,16 @@ fn should_report_from_block_count(block_count: u32) -> bool {
     block_count > 0 && block_count % BLOCK_REPORTING_FREQ_NUM == 0
 }
 
-impl std::fmt::Debug for IndexerState {
+impl std::fmt::Display for IndexerState {
```
Comment (jenr24G) [inline src/state/mod.rs:980]: porque no los dos?

Comment (jenr24G) [reply src/state/mod.rs:980]: more detail but less pretty for `Debug`

**Review by jenr24G — APPROVED** [review]
Looks great! Ready for merge pending CI checks!

File: README.md (hunk 1 above)
Comment (Isaac-DeFrain) [reply README.md:40]: I could've sworn you said this was all that was required... Never mind, will revert

File: src/state/mod.rs (hunk 5 above)
Comment (Isaac-DeFrain) [reply src/state/mod.rs:980]: Meant to derive `Debug`! Added.

File: src/state/branch.rs (hunk 3 above)
Comment (Isaac-DeFrain) [reply src/state/branch.rs:78]: Wow vestigial code from the past that was present in `snapshot` and got included in the cherry-pick! Removed.

**Pull request merged**